File: go-version/leetcode/LC1290/get_decimal_value.go
package LC1290

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/7/14 下午11:24
 * @FilePath: get_decimal_value
 * @Description:
 */

// ListNode Definition for singly-linked list.
type ListNode struct {
	Val  int
	Next *ListNode
}

func getDecimalValue(head *ListNode) int {
	node := head
	ans := 0
	for node != nil {
		ans *= 2
		ans += node.Val
		node = node.Next
	}
	return ans
}

File: python-version/LC1290/get_decimal_value.py
from typing import Optional


# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    @staticmethod
    def getDecimalValue(head: Optional[ListNode]) -> int:
        node = head
        ans = 0
        while node:
            ans *= 2
            ans += node.val
            node = node.next
        return ans
